dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/validation/service/attribute/TrackedAttributeValidationService.java

```diff
@@ -1,164 +0,0 @@
- * specific prior written permission.
- *
- * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
- * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
- * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
- * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
- * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
- * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
- * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
- * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
- * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
- * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
- */
-package org.hisp.dhis.tracker.imports.validation.service.attribute;
-
-import java.util.List;
-import java.util.Optional;
-import java.util.function.Predicate;
-import lombok.Getter;
-import lombok.RequiredArgsConstructor;
-import org.apache.commons.lang3.StringUtils;
-import org.hisp.dhis.common.ValueType;
-import org.hisp.dhis.fileresource.FileResource;
-import org.hisp.dhis.fileresource.FileResourceService;
-import org.hisp.dhis.system.util.MathUtils;
-import org.hisp.dhis.trackedentity.TrackedEntityAttribute;
-import org.hisp.dhis.tracker.imports.util.Constant;
-import org.hisp.dhis.user.UserService;
-import org.hisp.dhis.util.DateUtils;
-import org.springframework.stereotype.Component;
-import org.springframework.transaction.annotation.Transactional;
-
-/**
- * @author Luca Cambi
- */
-@Component
-public class TrackedAttributeValidationService {
-  /**
-   * @author Luca Cambi
-   */
-  @Getter
-  @RequiredArgsConstructor
-  private static class ValueTypeValidationFunction {
-    private final ValueType valueType;
-
-    private final Predicate<String> function;
-
-    private final String message;
-  }
-
-  private static final String VALUE_STRING = "Value";
-
-  private final FileResourceService fileResourceService;
-
-  private final List<ValueTypeValidationFunction> valueTypeValidationFunctions;
-
-  public TrackedAttributeValidationService(
-      UserService userService, FileResourceService fileResourceService) {
-    this.fileResourceService = fileResourceService;
-    valueTypeValidationFunctions =
-        List.of(
-            new ValueTypeValidationFunction(
-                ValueType.NUMBER,
-                v -> !MathUtils.isNumeric(v),
-                " '%s' is not a valid numeric type for attribute %s "),
-            new ValueTypeValidationFunction(
-                ValueType.BOOLEAN,
-                v -> !MathUtils.isBool(v),
-                " '%s' is not a valid boolean type for attribute %s "),
-            new ValueTypeValidationFunction(
-                ValueType.DATE,
-                v -> DateUtils.parseDate(v) == null,
```

Comment:
where did these validations go?

Reply:
These vaidations were completely overlapping the validation of `ValueType` done [here](https://github.com/dhis2/dhis2-core/blob/master/dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/util/ValidationUtils.java#L515-L539)

Reply:
but where do we call these?

Reply:
[Here](https://github.com/dhis2/dhis2-core/blob/7ecb3576230c66886c48d23042b92fb409891061/dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/validation/validator/ValidationUtils.java#L262)
We call the common value type validation that is the same for the whole system and it is covering the validations that were deleted.